### ISSUE TITLE
TTS problem not analysed correctly

### ISSUE THREAD

**Issue by emeschi** [cmsdaq/DAQExpert#55]
Overnight there was an instance where several partitions reported disconnected. This is the relevant ELOG message:
"This is the message shown by DAQ expert :
TTCP TEC- in TRACKER subsystem is in disconnected TTS state. It's blocking trigger.The PI of the subsystem may be suffering from a firmware problem
Due to this TEC- disconnected state Running state is "trigger blocked by TTS" in TTS and trigger blockers page."

This analysis is incorrect, at least and for sure for legacy FED partitions. The PI may just be right in reporting this state - the input states from the FMMs should be checked... Further evidence is if there are other partitions in the same state which indicates there is likely a problem with the FMM crate instead (it was the case and three crates had to be power cycled). A problem with the TCDS could only be suspected if the OR of the FMM states is different from what the PI reports.
If the problem could be clearly identified we could actually give instructions to the shifter to try and fix it

**Comment by gladky** [comment]
Thank you for reporting this issue. After a discussion with Hannes we propose following analysis in case of partition disconnected:

```
IF: legacy partition
    IF: fmm disconnected
        IF: any input of FMM disconnected
            return FED-DISCONNECTED
        ELSE:
            return FMM-PROBLEM
    ELSE:
        return PI-PROBLEM
ELSE
    return PI-DISCONNECTED
```

| case | message | suggestion |
|---|---|---|
| FED-DISCONNECTED | FED(s) in disconnected: [list]  | |
| FMM-PROBLEM |FMM problem. FMM gives disconnected output but no input is disconnected | Call DAQ on call to powercycle FMM crate |
| PI-PROBLEM |PI problem ||
| PI-DISCONNECTED | PI is disconnected, problem with PI or its input. Cannot investigate more - no PI input| |

**Comment by gladky** [comment]
I have rerun the expert with improved disconnect analysis logic on development instance. This is the result for saturday case reported by @emeschi:

> **FMM problem**
> FMM with url http://fmmpc-s1d12-07-01.cms:11100 and geoslot 3 of partition TEC- in TRACKER subsystem has output state disconnected but no input is disconnected.
> _Duration: 24:38 25 minutes_
> 
> **Action:**
> Call DAQ on-call to powercycle FMM crate

You can check the results for this case in expert browser here:
http://daq-expert-dev.cms/DAQExpert/?start=2017-04-01T01:15:03.752Z&end=2017-04-01T02:18:58.106Z

Please let me know if you have further suggestions to this analysis - especially what suggestions do we have for shifters in case of other subcases from table above (FED-DISCONNECTED, PI-PROBLEM, PI-DISCONNECTED).

**Comment by gladky** [comment]
All cases covered with tests. Released with [2.3.0](https://github.com/cmsdaq/DAQExpert/tree/release/2.3.0)
